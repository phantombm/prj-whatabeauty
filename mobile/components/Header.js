import React, { Component } from 'react';
import { Text, View, TouchableWithoutFeedback } from 'react-native';
import PropTypes from 'prop-types';
import { Ionicons } from '@expo/vector-icons';

export default class Header extends Component {
  static propTypes = {
    onPressIcon: PropTypes.func,
    icon: PropTypes.element,
    title: PropTypes.string
  };

  static defaultProps = {
    icon: <Ionicons name="md-menu" size={ 32 } />
  };

  // TODO: this 'add' method is for testing jest unit test
  add = (a, b) => {
    return a + b;
  };

  render() {
    return (
      <View style={{ flex: 1, flexDirection: 'row' }}>
        <View style={{ flex: 1 }}>
          <TouchableWithoutFeedback onPress={ this.props.onPressIcon }>
            <View style={{
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center'
            }}>{ this.props.icon }</View>
          </TouchableWithoutFeedback>
        </View>
        <View style={{
          flex: 4,
          alignItems: 'center',
          justifyContent: 'center'
        }}>
          <Text>{ this.props.title }</Text>
        </View>
        <View style={{ flex: 1 }}></View>
      </View>
    );
  }
}
