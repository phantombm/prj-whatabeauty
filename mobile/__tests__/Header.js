import React from 'react';
import 'react-native';
import testRenderer from 'react-test-renderer';

import Header from '../components/Header';

test('renders correctly', () => {
  const tree = testRenderer.create(
    <Header />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('add', () => {
  const header = new Header();

  expect(header.add(5 , 10)).toBe(15);
  expect(header.add(-5 , 10)).toBe(5);
  expect(header.add(5 , -10)).toBe(-5);
  expect(header.add(-5 , -10)).toBe(-15);
});
