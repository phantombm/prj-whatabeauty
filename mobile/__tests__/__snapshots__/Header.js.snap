// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
